x/consensus/module.go

```diff
@@ -0,0 +1,182 @@
+package consensus
+
+import (
+	"context"
+	"encoding/json"
+
+	"cosmossdk.io/core/appmodule"
+	"cosmossdk.io/depinject"
+
+	gwruntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	modulev1 "cosmossdk.io/api/cosmos/consensus/module/v1"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	store "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/consensus/exported"
+	"github.com/cosmos/cosmos-sdk/x/consensus/keeper"
+	"github.com/cosmos/cosmos-sdk/x/consensus/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+// ConsensusVersion defines the current x/bank module consensus version.
+const ConsensusVersion = 1
```

Comment:
note: the v2 migrations will never run if the ConsensusVersion here is 1

Reply:
could you explain why?

Reply:
The upgrade handlers run each module's migrations from version A to B, where:
- A is the module's version stored in the x/upgrade state
- B is the module's consensusVersion defined in this file

Here: A=0 (it's a new module), and B=1 (this line). So it will run `Migrate0To1` (which is InitGenesis, see https://github.com/cosmos/cosmos-sdk/pull/13164#discussion_r964842527), not `Migrate1to2`.